ui/plugins/eu.esdihumboldt.hale.ui.functions.geometric/src/eu/esdihumboldt/hale/ui/functions/geometric/CoordinateReferenceSystemParameterPage.java

```diff
@@ -0,0 +1,136 @@
+package eu.esdihumboldt.hale.ui.functions.geometric;
+
+import java.util.List;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.DialogPage;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.geotools.referencing.CRS;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+
+import eu.esdihumboldt.cst.functions.geometric.ReprojectGeometryFunction;
+import eu.esdihumboldt.hale.common.align.extension.function.PropertyFunctionExtension;
+import eu.esdihumboldt.hale.common.align.model.ParameterValue;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.CodeDefinition;
+import eu.esdihumboldt.hale.common.instance.geometry.impl.WKTDefinition;
+import eu.esdihumboldt.hale.common.schema.geometry.CRSDefinition;
+import eu.esdihumboldt.hale.ui.function.generic.pages.AbstractParameterPage;
+import eu.esdihumboldt.hale.ui.io.instance.crs.SelectCRSDialog;
+
+public class CoordinateReferenceSystemParameterPage extends AbstractParameterPage implements ReprojectGeometryFunction {
+
+	private Label crsLabel;
+	private Button selectCrs;
+	private CRSDefinition crsDef;
+	
+	public CoordinateReferenceSystemParameterPage() {
+		super(PropertyFunctionExtension.getInstance().get(ID), "Please select the CRS of the destination geometry");
+		setPageComplete(false);
+	}
+
+	@Override
+	protected void onShowPage(boolean firstShow) {
+		super.onShowPage(firstShow);
+
+		if (firstShow && getWizard().getInitCell() != null) {
+			List<ParameterValue> parameters = getWizard().getInitCell()
+					.getTransformationParameters()
+					.get(PARAMETER_REFERENCE_SYSTEM);
+			if (parameters != null && parameters.size() > 0) {
+				String crs = parameters.get(0).getStringRepresentation();
+				if (crs.startsWith("EPSG:")) {
+					crsDef = new CodeDefinition(crs, null);
+				} else {
+					crsDef = new WKTDefinition(crs, null);
+				}
+			}
+		}
+		
+		update();
+	}
+
+	@Override
+	public ListMultimap<String, ParameterValue> getConfiguration() {
+		ListMultimap<String, ParameterValue> conf = ArrayListMultimap.create();
+		
+		String crsAsString = null;
+		if (crsDef != null) {
+			if (crsDef instanceof CodeDefinition) {
+				crsAsString = ((CodeDefinition)crsDef).getCode();
+			} else if (crsDef instanceof WKTDefinition) {
+				crsAsString = ((WKTDefinition)crsDef).getWkt();
+			}
+		}
+		conf.put(PARAMETER_REFERENCE_SYSTEM, new ParameterValue(crsAsString));
```

Comment:
Thanks for the suggestion, I wasn't aware of the existence and purpose of `CRSDefinitionManager`. I implemented the change and the code runs fine in the HALE product, but for some reason `ReprojectGeometryTest` fails when the Tests product is run. After debugging, I noticed that `CRSDefinitionManager.getInstance().parse(crsString)` returns null, even if `crsString = "EPSG:4326"`. I guess there is something wrong with the lookup of OSGi services in `CRSDefinitionManager.getDefinitions()`, but since I'm a complete beginner in OSGi development I can't figure out what... any ideas?